### PR TITLE
feat: Add preference validation tests 

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

The functests are refactored and tests that validate preferences by
booting guests with the according OS are added. The added tests can be
skipped or focussed by the added FUNCTEST_EXTRA_ARGS variable, which
can be defined when running the cluster-functest or kubevirt-functest
targets.

E.g. run the following to skip all preference validation tests:

FUNCTEST_EXTRA_ARGS='--ginkgo.skip="VirtualMachine using a preference is able to boot"' make cluster-functest

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by jcanocan** [comment]
This is a nice refactor job! Thanks a lot! /lgtm

File: tests/functests/instancetype_test.go [hunk 1]
```diff
@@ -2,23 +2,58 @@ package tests
...
 	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+	"kubevirt.io/client-go/kubecli"
 )
 
-const vmReadyTimeout = 300 * time.Second
+const (
+	vmReadyTimeout = 300 * time.Second
+	sshPort        = 22
+
+	fedoraContainerDisk        = "quay.io/containerdisks/fedora:latest"
```
Comment (codingben) [inline tests/functests/instancetype_test.go:27]: Why do you use latest tag?

File: tests/functests/instancetype_test.go [hunk 2]
```diff
@@ -50,81 +76,175 @@ var _ = Describe("Common instance types func tests", func() {
...
+			privKey := addCloudInitWithAuthorizedKey(vm)
+			vm, err = virtClient.VirtualMachine(testNamespace).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+			expectVMToBeReady(virtClient, vm.Name)
+			if guestAgent {
+				expectGuestAgentToBeConnected(virtClient, vm.Name)
+			}
+			expectSSHToRunCommandWithPrivKey(virtClient, vm.Name, username, privKey)
+		},
+			Entry("Fedora", fedoraContainerDisk, "fedora", "fedora", true),
```
Comment (codingben) [inline tests/functests/instancetype_test.go:100]: Same here, what is the reason of not using a specific Fedora version? e.g. Fedora 39?

File: tests/functests/instancetype_test.go [hunk 3]
```diff
@@ -50,81 +76,175 @@ var _ = Describe("Common instance types func tests", func() {
...
+			Entry("CentOS 7", centos7ContainerDisk, "centos.7", "centos", true),
+			Entry("CentOS Stream 8", centosStream8ContainerDisk, "centos.stream8", "centos", true),
+			Entry("CentOS Stream 9", centosStream9ContainerDisk, "centos.stream9", "cloud-user", true),
+			Entry("Ubuntu 18.04", ubuntu1804ContainerDisk, "ubuntu", "ubuntu", false),
+			Entry("Ubuntu 20.04", ubuntu2004ContainerDisk, "ubuntu", "ubuntu", false),
+			Entry("Ubuntu 22.04", ubuntu2204ContainerDisk, "ubuntu", "ubuntu", false),
```
Comment (codingben) [inline tests/functests/instancetype_test.go:106]: What is the reason of testing multiple Ubuntu versions? It has costs of time duration during execution of the tests.

**Review by codingben — APPROVED** [review]
/lgtm

File: tests/functests/instancetype_test.go (hunk 1 above)
Comment (0xFelix) [reply tests/functests/instancetype_test.go:27]: Fedora is a relatively fast moving target, as such in my opinion it makes sense to always test the latest available release.

File: tests/functests/instancetype_test.go (hunk 2 above)
Comment (0xFelix) [reply tests/functests/instancetype_test.go:100]: Same as above

File: tests/functests/instancetype_test.go (hunk 3 above)
Comment (0xFelix) [reply tests/functests/instancetype_test.go:106]: IMO it makes sense to test all long term releases, as long as they are supported.

**Pull request merged**